templates/experiments/single_experiment_home.html

```diff
@@ -212,54 +212,244 @@ <h3 class="font-bold text-lg">Parent Experiments:
+              {{ experiment_versions|default:"[]"|json_script:"experiment-versions-data" }}
               {% if perms.experiments.download_chats %}
                 {% include "experiments/components/exports.html" %}
               {% endif %}
-              <form
-                hx-trigger="load, change"
-                hx-target="#sessions-table"
-                hx-get="{% url 'experiments:sessions-list' request.team.slug experiment.id %}"
-                hx-swap="innerHTML"
-              >
-                <div class="flex gap-2">
-                  <div class="w-72 my-2">
-                    <input x-model="participants" name="participants" id="participants-filter" placeholder="Filter participants...">
+              <div x-data="filterComponent" @click.outside="filterData.showFilters = false">
+              <!-- Filter Button -->
+                <div class="flex items-center">
+                  <button @click="toggleFilters()" class="btn btn-sm btn-outline">
+                    <i class="fa-solid fa-filter"></i> Filter <span x-show="getActiveFilterCount() > 0" x-cloak>(<span x-text="getActiveFilterCount()"></span>)</span>
+                  </button>
+
+                  <div x-show="filterData.loading" x-cloak class="ml-2">
+                    <i class="fa-solid fa-spinner fa-spin"></i>
                   </div>
-                  {% if perms.annotations.view_customtaggeditem %}
-                    <div class="w-72 my-2">
-                      <select x-model="tags" id="tag-multiselect-filter" name="tags" multiple
-                              placeholder="Filter tags..."
-                              autocomplete="off">
-                        <option value="">Filter tags...</option>
-                        {% for tag in available_tags %}
-                          <option value="{{ tag.name }}">{{ tag.name }}</option>
-                        {% endfor %}
-                      </select>
-                    </div>
-                  {% endif %}
-                  <div class="form-control">
-                    <label class="label cursor-pointer">
-                      <span class="label-text mr-2">Include API Sessions</span>
-                      <input type="checkbox" name="show-all" class="toggle" />
-                    </label>
+
+                  <!-- Clear filters button -->
+                  <button
+                    x-show="getActiveFilterCount() > 0 && !filterData.loading"
+                    @click="clearAllFilters()"
+                    class="btn btn-square btn-sm btn-ghost"
+                    title="Clear all filters" x-cloak>
+                    x
+                  </button>
+                </div>
+
+              <!-- Filters Panel -->
+                <div
+                  x-show="filterData.showFilters"
+                  class="absolute left-0 mt-2 p-4 bg-white border border-gray-200 rounded-lg shadow-lg z-10 w-fit"
+                >
+                  <div class="space-y-3">
```

Comment:
nit: make it more compact
```suggestion
                  <div class="space-y-2">
```